### PR TITLE
video upload works

### PR DESCRIPTION
**Pull request by jtaylor-orbis**
### TL;DR

Added video upload functionality and improved ping message handling in the LINE bot.

### What changed?

- Introduced new environment variables: `CF_STREAM_TOKEN` and `CF_ACCOUNT_ID` for Cloudflare Stream integration.
- Implemented video upload functionality using Cloudflare Stream API.
- Enhanced message processing to differentiate between ping and video messages.
- Improved ping message handling with more robust parsing and error handling.
- Updated the response structure for different types of messages.

### How to test?

1. Set up the new environment variables: `CF_STREAM_TOKEN` and `CF_ACCOUNT_ID`.
2. Send a ping message in the format: "ping: [message].[location]"
3. Send a video message in the format: "video: [video_url].[coordinates]"
4. Verify that ping messages are processed correctly and return location information.
5. Confirm that video messages trigger the upload to Cloudflare Stream and return appropriate responses.
6. Test error handling by sending incorrectly formatted messages.

### Why make this change?

This change enhances the bot's capabilities by adding video upload functionality and improving the existing ping feature. It allows users to share both location-based information and video content, making the bot more versatile and user-friendly. The improved message handling also ensures better error management and more accurate responses to user inputs.

### CODE REVIEW

**Comment by jtaylor-orbis** [comment]
* **#20** <a href="https://app.graphite.com/github/pr/orbisoperations/catalyst-linebot/20?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#19** <a href="https://app.graphite.com/github/pr/orbisoperations/catalyst-linebot/19?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.com/github/pr/orbisoperations/catalyst-linebot/19?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `main`


<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

File: src/index.ts [hunk 1]
```diff
@@ -165,92 +167,145 @@ app.use('/', async (c) => {
...
-					limit: '1'
-				}),
-					{
-						method: "GET",
-						headers: {
-							'X-RapidAPI-Key': c.env.RAPID_API_KEY,
-							'X-RapidAPI-Host': 'maptoolkit.p.rapidapi.com'
+			// add routine to check video upload
+
+			const [pingSelector, videoSelector] = ["ping:", "video:"]
```
Comment (jaeyojae) [inline src/index.ts:174]: I'm not liking this, would like to consider how to make more dynamic. et's add this req for the user in the README

File: src/index.ts [hunk 2]
```diff
@@ -165,92 +167,145 @@ app.use('/', async (c) => {
...
+
+			const [isPing, isVideo] = [
+				message.message.text.toLowerCase().startsWith(pingSelector),
+				message.message.text.toLowerCase().startsWith(videoSelector)
+			]
+
+			console.log("isPing: ", isPing, "isVideo: ", isVideo)
+			if (isPing) {
```
Comment (jaeyojae) [inline src/index.ts:182]: In ideal world, would break these down and add some test cases try and find the edge cases here.

**Review by jaeyojae — APPROVED** [review]

**Review by jaeyojae — APPROVED** [review]

**Comment by ihammers-orbis** [comment]
Closing stale PR as part of adapter handoff cleanup
